src/System.Net.Security/src/System/Net/Security/Pal.OSX/SafeDeleteSslContext.cs

```diff
@@ -0,0 +1,345 @@
+            }
+            catch (Exception ex)
+            {
+                Debug.Write("Exception Caught. - " + ex);
+                Dispose();
+                throw;
+            }
+        }
+
+        private static SafeSslHandle CreateSslContext(SafeFreeSslCredentials credential, bool isServer)
+        {
+            switch (credential.Policy)
+            {
+                case EncryptionPolicy.RequireEncryption:
+                case EncryptionPolicy.AllowNoEncryption:
+                    // SecureTransport doesn't allow TLS_NULL_NULL_WITH_NULL, but
+                    // since AllowNoEncryption intersect OS-supported isn't nothing,
+                    // let it pass.
+                    break;
+                default:
+                    throw new PlatformNotSupportedException(SR.net_encryptionpolicy_notsupported);
+            }
+
+            SafeSslHandle sslContext = Interop.AppleCrypto.SslCreateContext(isServer ? 1 : 0);
+
+            try
+            {
+                if (sslContext.IsInvalid)
+                {
+                    // This is as likely as anything.  No error conditions are defined for
+                    // the OS function, and our shim only adds a NULL if isServer isn't a normalized bool.
+                    throw new OutOfMemoryException();
+                }
+
+                // Let None mean "system default"
+                if (credential.Protocols != SslProtocols.None)
+                {
+                    SetProtocols(sslContext, credential.Protocols);
+                }
+
+                if (credential.Certificate != null)
+                {
+                    SetCertificate(sslContext, credential.Certificate);
+                }
+
+                Interop.AppleCrypto.SslBreakOnServerAuth(sslContext, true);
+                Interop.AppleCrypto.SslBreakOnClientAuth(sslContext, true);
+            }
+            catch
+            {
+                sslContext.Dispose();
+                throw;
+            }
+
+            return sslContext;
+        }
+
+        public override bool IsInvalid => _sslContext?.IsInvalid ?? true;
+
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                if (null != _sslContext)
+                {
+                    _sslContext.Dispose();
+                    _sslContext = null;
+                }
+
+                _toConnection = null;
+                _fromConnection = null;
+                _writeCallback = null;
+                _readCallback = null;
+            }
+
+            base.Dispose(disposing);
+        }
+
+        private unsafe int WriteToConnection(void* connection, byte* data, void** dataLength)
+        {
+            ulong toWrite = (ulong)*dataLength;
+            byte* readFrom = data;
+
+            lock (_toConnection)
+            {
+                while (toWrite > 0)
+                {
+                    _toConnection.Enqueue(*readFrom);
```

Comment:
Yeah, I went with Queue so I didn't have to manage my own growing/shrinking producer/consumer model.  It would be better with a locally managed circular buffer which is optimized for/by Buffer.BlockCopy; but with everything else that was changing I went with the KISS paradigm on the data transfer between native and managed.